pkg/activator/net/throttler_test.go

```diff
@@ -65,130 +65,199 @@ func newTestThrottler(ctx context.Context) *Throttler {
-	// but check that we behave reasonably if it's exceeded.
-	capacity := rt.calculateCapacity(revisionMaxConcurrency+5, 1)
-	if got, want := capacity, queue.MaxBreakerCapacity; got != want {
-		t.Errorf("calculateCapacity = %d, want: %d", got, want)
-	}
-
-	// Now test with podIP trackers in tow.
-	// Simple case.
-	rt.numActivators.Store(1)
-	rt.activatorIndex.Store(0)
-	rt.podTrackers = makeTrackers(1, 10)
-	rt.containerConcurrency = 10
-	rt.breaker = queue.NewBreaker(testBreakerParams)
-
-	rt.updateCapacity(0 /* doesn't matter here*/)
-	if got, want := rt.breaker.Capacity(), 10; got != want {
-		t.Errorf("Capacity = %d, want: %d", got, want)
-	}
-
-	// 2 backends.
-	rt.podTrackers = makeTrackers(2, 10)
-	rt.updateCapacity(-1 /* doesn't matter here*/)
-	if got, want := rt.breaker.Capacity(), 20; got != want {
-		t.Errorf("Capacity = %d, want: %d", got, want)
-	}
-
-	// 2 activators.
-	rt.numActivators.Store(2)
-	rt.updateCapacity(-1 /* doesn't matter here*/)
-	if got, want := rt.breaker.Capacity(), 10; got != want {
-		t.Errorf("Capacity = %d, want: %d", got, want)
-	}
 
-	// 3 pods, index 0.
-	rt.podTrackers = makeTrackers(3, 10)
-	rt.updateCapacity(-1 /* doesn't matter here*/)
-	if got, want := rt.breaker.Capacity(), 15; got != want {
-		t.Errorf("Capacity = %d, want: %d", got, want)
+	tests := []struct {
+		name                 string
+		capacity             int
+		numActivators        int32
+		activatorIndex       int32
+		containerConcurrency int
+		isNewInfiniteBreaker bool
+		podTrackers          []*podTracker
+		want                 int
+		checkAssignedPod     bool
+	}{
+		{
+			name:                 "capacity: 1, cc: 10",
+			capacity:             1,
+			containerConcurrency: 10,
+			want:                 1 * 10,
+		},
+		{
+			name:                 "capacity: 10, cc: 10",
+			capacity:             10,
+			containerConcurrency: 10,
+			want:                 10 * 10,
+		},
+		{
+			name:                 "capacity: 100, cc: 10",
+			capacity:             100,
+			containerConcurrency: 10,
+			want:                 100 * 10,
+		},
+		{
+			name:                 "numActivators: 5, capacity: 10, cc: 10",
+			capacity:             10,
+			numActivators:        5,
+			containerConcurrency: 10,
+			want:                 10 * 10 / 5,
+		},
+		{
+			name:                 "numActivators: 200, capacity: 10, cc: 10",
+			capacity:             10,
+			numActivators:        200,
+			containerConcurrency: 10,
+			want:                 1,
+		},
+		{
+			name:                 "numActivators: 200, capacity: 0, cc: 10",
+			capacity:             0,
+			numActivators:        200,
+			containerConcurrency: 10,
+			want:                 0,
+		},
+		// now test with CC=0.
+		// in the CC=0 cases we use the infinite breaker whose capacity can either be
+		// totally blocked (0) or totally open (1).
+		{
+			name:                 "newInfiniteBreaker, numActivators: 200, capacity: 0, cc: 0",
```

Comment:
I think the test earlier forgot to reset the num activators to zero - can we add that and the extra cases